File: C++/Leetcode/Two Pointers/isSubsequence.cpp
#include <iostream>
#include <string>

class Solution {
    public:
        bool isSubsequence(std::string s, std::string t){
            if (s == ""){
                return true;
            }
            else if (t == ""){
                return false;
            }
            else {
                if (s[0] == t[0]){
                    return isSubsequence(s.substr(1, s.size()), t.substr(1, t.size()));
                }
                else {
                    return isSubsequence(s, t.substr(1, t.size()));
                }
            }
        }
};

int main(){
    Solution solution_instance;
    std::cout << solution_instance.isSubsequence("abc", "ahbgdc") << std::endl;
}

File: Python/Leetcode/Two Pointers/isSubsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        if not s: 
            return True
        l = 0
        for i in range(len(t)):
            if l == len(s):
                return True
            elif s[l] == t[i]:
                l += 1
        
        return l == len(s)
            
if __name__ == '__main__':
    solution_instance = Solution()
    print(solution_instance.isSubsequence("axc", "ahbgdc"))